command/agent/http.go

```diff
@@ -193,6 +194,58 @@ func NewHTTPServers(agent *Agent, config *Config) ([]*HTTPServer, error) {
 		srvs = append(srvs, srv)
 	}
 
+	// Initialize the API socket if configured
+	startAPISocket := func() {
+		if config.APISocket.Path == "" {
+			return
+		}
+
+		const (
+			MaxSocketPathLength = 104
+			msgPrefix           = "error starting api socket:"
+		)
+
+		if len(config.APISocket.Path) > MaxSocketPathLength {
+			serverInitializationErrors = multierror.Append(serverInitializationErrors,
+				fmt.Errorf("%s path must be %v characters or less", msgPrefix, MaxSocketPathLength))
+		}
+
+		sl, err := listenerutil.UnixSocketListener(config.APISocket.Path, config.APISocket.UnixSocketsConfig())
```

Comment:
If we're going to use this library we should make sure we've fully documented the expected behavior for users. You can set `user` by name or ID, but `group` only by ID (well, that's how it's _documented_ but not how it behaves if you look at the source). If you set `path` without setting `user`, `group`, and `mode`, the socket is owned by the Nomad agent's user and the mode will be determined by the umask (which is usually different for login shells vs services, etc.)

Maybe we should have an explicit default configuration here rather than relying on the default from the library? i feel like that's likely to get changed out from under us.

Reply:
+1 on observing the differences in implementation and documentation on that function. What are your thoughts on a reasonable default configuration, or are you saying that we should just codify the current defaults in Nomad so that if they _do change in the library_ that we won't be surprised/broken by it?

Reply:
The current defaults from the library seem reasonable, so let's codify those defaults in Nomad so that we control them going forward.

Reply:
I can ensure the behavior around user and group by making the defaults concrete, but the behavior around `mode` is a moving target since it's solely determined by how the library function handles that argument. I did add a test to validate whether or not the behavior of the library call has changed from we're expecting. Seem okay?